queue_processor/queue_worker.py

```diff
@@ -0,0 +1,348 @@
+            getting_json_time = time.time() - getting_json_time
+            logging.info("Got JSON in %.2f seconds", getting_json_time)
+            self.dd_stats.timing(self.dd_prefix + "get_json", getting_json_time)
+
+            # pick out errors and write them to the DB
+            paper_id_to_error = {}
+            for line in dataprep2.json_from_file(json_file_name):
+                if "error" not in line:
+                    continue
+                error = line["error"]
+                error["message"] = dataprep2.sanitize_for_json(error["message"])
+                error["stackTrace"] = dataprep2.sanitize_for_json(error["stackTrace"])
+                paper_id = error["docName"]
+                if paper_id.endswith(".pdf"):
+                    paper_id = paper_id[:-4]
+                paper_id_to_error[paper_id] = error
+                logging.error("Paper %s has error: %s", paper_id, error["message"])
+
+            self.dd_stats.increment(self.dd_prefix + "errors", len(paper_id_to_error))
+
+            # make unlabeled tokens file
+            logging.info("Making unlabeled tokens ...")
+            making_unlabeled_tokens_time = time.time()
+            unlabeled_tokens_file_name = os.path.join(temp_dir.name, "unlabeled-tokens.h5")
+            dataprep2.make_unlabeled_tokens_file(
+                json_file_name,
+                unlabeled_tokens_file_name,
+                ignore_errors=True)
+            os.remove(json_file_name)
+            making_unlabeled_tokens_time = time.time() - making_unlabeled_tokens_time
+            logging.info("Made unlabeled tokens in %.2f seconds", making_unlabeled_tokens_time)
+            self.dd_stats.timing(self.dd_prefix + "make_unlabeled", making_unlabeled_tokens_time)
+
+            # make featurized tokens file
+            logging.info("Making featurized tokens ...")
+            making_featurized_tokens_time = time.time()
+            with h5py.File(unlabeled_tokens_file_name, "r") as unlabeled_tokens_file:
+                featurized_tokens_file_name = os.path.join(temp_dir.name, "featurized-tokens.h5")
+                dataprep2.make_featurized_tokens_file(
+                    featurized_tokens_file_name,
+                    unlabeled_tokens_file,
+                    token_stats,
+                    embeddings,
+                    dataprep2.VisionOutput(None),
+                    model_settings
+                )
+            # We don't delete the unlabeled file here because the featurized one contains references to it.
+            making_featurized_tokens_time = time.time() - making_featurized_tokens_time
+            logging.info("Made featurized tokens in %.2f seconds", making_featurized_tokens_time)
+            self.dd_stats.timing(self.dd_prefix + "make_featurized", making_featurized_tokens_time)
+
+            yield temp_dir, featurized_tokens_file_name, sha_to_paper_id_and_messages
+
+    def _get_sha_to_paper_id_and_messages(self):
```

Comment:
I think the data flow is inverted here, and the `QueueWorker` seems to have a lot of methods that don't really have to do with processing messages from the upstream queue. I think it would make more sense if the logic in this function was lifted to the `main` in `process_queue.py`, and, for each message pulled from the queue, parse it from an `SQS.Message` to some `InternalMessage` shape, then pass it to some other function to processed.

Here's an example [in this repo](https://github.com/allenai/pdf-extraction/blob/master/parse-merge/src/main/scala/ParseMergeWorker.scala#L47-L58), and a more complicated one [in scholar](https://github.com/allenai/scholar/blob/master/pipeline/src/main/scala/org/allenai/s2/pipeline/spark/steps/incremental/ClusterSourcedPapers.scala#L71).

Reply:
Agreed. This data flow was inherited from the original code, except that it uses `TaskDB` instead of queue. I want to refactor the data flow in subsequent PR in `pdf-extraction` repo.

Reply:
There'll be a few PRs to come because some things still need fixing. This is only a first cut, barely working version.

Reply:
Goooot it got it. Do you think it makes sense to not bring over these commit into `pdf-extraction`, then? I think it could make sense to have two PRs:

- PR 1: The Great Migration™️ 
- PR 2: Refactoring to read from a queue + some quality of life improvements.

Thoughts?

Reply:
👍